### ISSUE TITLE
type error on import due to misconfigured exports

### ISSUE THREAD

**Issue by OmarMcAdam** [davidje13/lean-qr#9]
getting an error when importing the package

```
Could not find a declaration file for module 'lean-qr'.
'.../node_modules/.pnpm/lean-qr@2.3.3/node_modules/lean-qr/build/index.mjs' 
implicitly has an 'any' type.

There are types at '.../apps/web-app/node_modules/lean-qr/index.d.ts', 
but this result could not be resolved when respecting package.json "exports".
The 'lean-qr' library may need to update its package.json or typings.
```

**Comment by davidje13** [comment]
Interesting; I don't fully understand the issue, but it [looks like TypeScript now requires separate `.d.cts` and `.d.mts` definition files?](https://github.com/microsoft/TypeScript/issues/52363) (for CJS / ESM respectively)

Can you share an MCVE for this issue? The lean-qr project [includes a test for typescript compatibility](https://github.com/davidje13/lean-qr/tree/master/test-package) (for esm only) which hasn't caught this, despite using TypeScript 5.3 (still the latest general-availability version). Are you using the pre-release 5.4 by any chance?

I also notice you're using ~Yarn PnP~ [pnpm](https://pnpm.io/), so I wonder if that could somehow be connected? - edit: I hadn't heard of pnpm before. I wonder if this is related to some difference in behaviour of that package manager?

(ideally I'm looking to add a test to ensure this stays fixed)

**Comment by NicoRumm** [comment]
It happens with `moduleResolution: "bundler"` in your `tsconfig.json` but not with `moduleResolution: "node"`

Should be easy to reproduce, but here is a minimal example anyway: https://github.com/NicoRumm/mre-lean-qr-9

**Comment by davidje13** [comment]
@NicoRumm thanks! Using that config replicated the issue, and I have published version 2.3.4 which resolves it (and includes a test to avoid the issue reappearing in the future)
